app.js

```diff
@@ -27,12 +27,36 @@ if (getURLParameter('touch') === 'true') {
 // Check if HTML form is hardcoded or needs to be retrieved
 // note: when running this file in enketo-core-performance-monitor xform = 'null'
 if (xform && xform !== 'null') {
-    document.querySelector('.guidance').remove();
-    xform = /^https?:\/\//.test(xform) ? xform : `${location.origin}/${xform}`;
-    const transformerUrl = `http://${location.hostname}:8085/transform?xform=${xform}`;
-    fetch(transformerUrl)
-        .then((response) => response.json())
-        .then((survey) => {
+    (async () => {
+        const isRemote = /^https?:\/\//.test(xform);
+        const xformURL = isRemote ? xform : `${location.origin}/${xform}`;
+        const transformerUrl = `http://${location.hostname}:8085/transform?xform=${xformURL}`;
+
+        try {
+            document.querySelector('.guidance').remove();
+
+            /** @type {import('enketo-transformer').TransformedSurvey & { modifiedTime?: number } | null} */
+            let survey = null;
+
+            if (!isRemote) {
+                // This must be dynamically imported or it'll be included in the build
+                const localForms = (await import(`./${'forms'}.mjs`)).default;
+                const localForm = localForms[xform];
+
+                if (localForm != null) {
+                    survey = {
+                        form: localForm.html_form,
+                        model: localForm.xml_model,
+                    };
+                }
+            }
+
+            if (survey == null) {
+                const response = await fetch(transformerUrl);
+
+                survey = await response.json();
+            }
+
```

Comment:
I am puzzled how this change appears to have made the WHO form so much faster to load (even though `initializeForm` is slightly slower).